### ISSUE TITLE
[java] Private field is detected as public inside nested classes in interfaces

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1578]
**Affects PMD Version:** 6.10.0

**Rule:** No rule explicitly, but rules that evaluate the Private attribute of FieldDeclarations

**Description:**
The XPath 2.0 Expression `//FieldDeclaration[@Private=true()]` should find the second field named `serialVersionUID`, but it doesn't.

**Code Sample demonstrating the issue:**

``` java
public interface MyInterface {
    public static final Map<String, String> MAPPING = Collections.unmodifiableMap(new HashMap<String, String>() {
        private static final long serialVersionUID = 3855526803226948630L;
        {
            put("X", "10");
            put("L", "50");
            put("C", "100");
            put("M", "1000");
        }
    });
}
```

**Running PMD through:** Designer
